src/main/java/com/powsybl/openloadflow/ac/equations/AcEquationSystem.java

```diff
@@ -324,26 +328,33 @@ private static void createNonImpedantBranch(LfBranch branch, LfBus bus1, LfBus b
         }
     }
 
-    private static void createBranchActivePowerTargetEquation(LfBranch branch, DiscretePhaseControl.ControlledSide controlledSide,
-                                                              EquationSystem<AcVariableType, AcEquationType> equationSystem, EquationTerm<AcVariableType, AcEquationType> p) {
-        branch.getDiscretePhaseControl()
-            .filter(dpc -> branch.isPhaseControlled(controlledSide) && dpc.getMode() == Mode.CONTROLLER)
-            .ifPresent(dpc -> {
-                if (dpc.getUnit() == DiscretePhaseControl.Unit.A) {
-                    throw new PowsyblException("Phase control in A is not yet supported");
-                }
-                equationSystem.createEquation(branch.getNum(), AcEquationType.BRANCH_TARGET_P).addTerm(p);
-
-                // we also create an equation that will be used later to maintain A1 variable constant
-                // this equation is now inactive
-                LfBranch controller = dpc.getController();
-                EquationTerm<AcVariableType, AcEquationType> a1 = equationSystem.getVariable(controller.getNum(), AcVariableType.BRANCH_ALPHA1)
-                        .createTerm();
-                branch.setA1(a1);
-                equationSystem.createEquation(controller.getNum(), AcEquationType.BRANCH_TARGET_ALPHA1)
-                        .addTerm(a1)
-                        .setActive(false);
-            });
+    private static void createBranchActivePowerTargetEquation(LfBranch branch, LfBus bus1, LfBus bus2, EquationSystem<AcVariableType, AcEquationType> equationSystem,
```

Comment:
Why does it solves our problem indeed because we don't go into this method for an update. What was the issue ?

Reply:
The bug was because of an `EquationTerm` reused 2 times in 2 different equations. This was not obvious in the `Equation`design and a check was missing (I added it, see `Equation.java`fix) so this not surprising that at 2 places in the code an `EquationTerm` was associated to 2 `Equation` (for phase shifter flow control and also generator reactive flow control). An `EquationTerm` has a reverse link to its `Equation` which is set when term is added. When the term was added to the second `Equation`, the first link was overwritten and so on lost. The `EquationSystem` indexing rely on this `EquationTerm` to `Equation` link and as the link was incorrect, the index was not updated correctly.